detect/errorloaderdetector.py

```diff
@@ -1,38 +1,68 @@
+import os 
+
 import pandas as pd
 
+from dataset.table import Table, Source
 from .detector import Detector
 
 
 class ErrorsLoaderDetector(Detector):
     """
-    Detector that loads a table of constant errors from csv with the columns:
-        _tid_: entity ID
-        attribute: attribute in violation
-        in the format _tid_, attribute
+    Detector that loads a table of constant errors with the columns:
+        id_col: entity ID
+        attr_col: attribute in violation
+        in the format id_col, attr_col
+    Can load these erros from a csv file or a relational table.
     """
-    def __init__(self, fpath, name='ErrorLoaderDetector'):
+    def __init__(self, fpath=None, 
+                 db_engine=None, table_name=None, schema_name=None, 
+                 id_col="_tid_", attr_col="attribute", 
+                 name="ErrorLoaderDetector"):
         """
+        :param fpath: (str) Path to source csv file to load errors
+        :param db_engine: (DBEngine) Database engine object
+        :param table_name: (str) Relational table considered for loading errors
+        :param schema_name: (str) Schema in which :param table_name: exists
+        :param id_col: (str) ID column name
+        :param attr_col: (str) Attribute column name
         :param name: (str) name of the detector
-        :param fpath: (str) path to csv file to load errors from.
+
+        To load from csv file, :param fpath: must be specified.
+        To load from a relational table, :param db_engine:, and 
+        :param table_name: must be specified, optionally specifying 
+        :param schema_name:.
         """
         super(ErrorsLoaderDetector, self).__init__(name)
-        self.errors_df = pd.read_csv(fpath,
-                                     dtype={'_tid_': int, 'attribute': str},
-                                     encoding='utf-8')
-        if list(self.errors_df) != ['_tid_', 'attribute']:
-            raise Exception('Invalid input file for ErrorsLoaderDetector: {}'.format(fpath))
+        src = None
+        dataset_name = None
+        if fpath is not None:
+            # Get the file name without the extension from fpath
+            dataset_name = os.path.basename(fpath).split()[0]
```

Comment:
let's not import `os` just for a datset name that we never use... Just use `dataset_name='errors'` or something.

Reply:
Done and the PR is updated.